src/main/java/com/avispl/symphony/dal/communicator/other/genericserver/WebClientCommunicator.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2015-2021 AVI-SPL, Inc. All Rights Reserved.
+ */
+package com.avispl.symphony.dal.communicator.other.genericserver;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.avispl.symphony.api.dal.dto.monitor.ExtendedStatistics;
+import com.avispl.symphony.api.dal.dto.monitor.Statistics;
+import com.avispl.symphony.api.dal.error.ResourceNotReachableException;
+import com.avispl.symphony.api.dal.monitor.Monitorable;
+import com.avispl.symphony.dal.communicator.HttpCommunicator;
+import com.avispl.symphony.dal.util.StringUtils;
+
+/**
+ * This class checks accessible to a given URI then return the HTTP status code to Symphony
+ */
+public class WebClientCommunicator extends HttpCommunicator implements Monitorable {
+	/**
+	 * Map to store HTTP status code and its description.
+	 */
+	private static final Map<Integer, String> STATUS_CODE_MAP = new HashMap<>();
+	private static final Pattern HTTP_STATUS_CODE_PATTERN = Pattern.compile("(\\d{3})");
+	private static final String SPLIT_LINE = "\n";
+	private static final String URI_STATUS = "URI Status";
+	private static final String NOT_CONFIGURED = "Not Configured";
+
+	static {
+		STATUS_CODE_MAP.put(200, " OK");
+		STATUS_CODE_MAP.put(400, " Bad Request");
+		STATUS_CODE_MAP.put(403, " Forbidden");
+		STATUS_CODE_MAP.put(401, " Unauthorized");
+	}
+
+	/**
+	 * URI string that is used to check accessible.
+	 */
+	private String URI;
+
+	/**
+	 * Constructor
+	 */
+	public WebClientCommunicator() {
+		// WebClientCommunicator no-args constructor
+	}
+
+	/**
+	 * Retrieves {@code URI}
+	 *
+	 * @return String This returns the current URI
+	 */
+	public String getURI() {
+		return this.URI;
+	}
+
+	/**
+	 * Sets {@code URI}
+	 *
+	 * @param URI This is the URI to be set
+	 */
+	public void setURI(String URI) {
+		this.URI = URI;
+	}
+
+	/**
+	 * This method is called by Symphony to get the list of statistics to be displayed
+	 * {@inheritDoc}
+	 *
+	 * @return List<Statistics> This returns the list of statistics
+	 */
+	@Override
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		if (logger.isDebugEnabled()) {
+			logger.debug("Perform doGet() at host " + this.getHost());
+		}
+		final ExtendedStatistics extStats = new ExtendedStatistics();
+		final Map<String, String> stats = new HashMap<>();
+		String uriStatusMessage;
+		if (!StringUtils.isNullOrEmpty(this.URI)) {
+			try {
+				doGet(this.URI);
+				uriStatusMessage = generateResponseMessage(200);
+			} catch (Exception e) {
+				String errorMessage = e.getMessage();

```

Comment:
getMessage() may in theory return null, so calling .split() on it afterwards may result with an NPE error, maybe it can be checked first with default empty value, or checked for null before use

Reply:
Updated with null checking